### PR TITLE
Using gulp to organize CI workflow(Part 1)

### PR DESCRIPTION
**Pull request by qinezh**

### CODE REVIEW

File: tools/Deployment/gulpfile.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        process.chdir(cwd);
+    });
+}
+
+gulp.task("build", () => {
+    return exec("powershell", ["./build.ps1", "-prod"], "../../../docfx/");
+});
+
+gulp.task("e2eTest:choco", () => {
+    return exec("choco", ["install", "firefox", "--version=46.0.1", "-y"]);
```
Comment (vicancy) [inline tools/Deployment/gulpfile.js]: move version number to config?

File: tools/Deployment/gulpfile.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    childProcess.stderr.on('data', (data) => {
+        process.stderr.write(data.toString());
+    })
+    return promise.then(() => {
+        process.chdir(cwd);
+    });
+}
+
+gulp.task("build", () => {
+    return exec("powershell", ["./build.ps1", "-prod"], "../../../docfx/");
```
Comment (vicancy) [inline tools/Deployment/gulpfile.js]: make hardcoded docfx path configurable or as parameter?

File: tools/Deployment/gulpfile.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+});
+
+gulp.task("e2eTest:choco", () => {
+    return exec("choco", ["install", "firefox", "--version=46.0.1", "-y"]);
+});
+
+gulp.task("e2eTest:buildSeed", ["build", "e2eTest:choco"], () => {
+    return exec("../docfx/target/Release/docfx/docfx.exe", ["docfx.json"], "../../../docfx-seed/");
```
Comment (vicancy) [inline tools/Deployment/gulpfile.js]: same here. hard-coded path means the folder structure is dependent on CI folder settings.

File: tools/Deployment/gulpfile.js [hunk 4]
```diff
@@ -0,0 +1,62 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+"use strict";
+
+let path = require('path');
+let fs = require('fs');
+
+let gulp = require("gulp");
```
Comment (superyyrrzz) [inline tools/Deployment/gulpfile.js]: >"gulp" [](start = 19, length = 6)

unify single and double quotation?

**Comment by superyyrrzz** [comment]
:shipit:

**Comment by vicancy** [comment]
naming: config_gulp.json?

File: tools/Deployment/gulpfile.js [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+    childProcess.stderr.on('data', (data) => {
+        process.stderr.write(data.toString());
+    })
+    return promise.then(() => {
+        process.chdir(cwd);
+    });
+}
+
+gulp.task("build", () => {
+    return exec("powershell", ["./build.ps1", "-prod"], config.docfx["home"]);
```
Comment (vicancy) [inline tools/Deployment/gulpfile.js]: provide a default value or throw exception when the config is not found?

**Comment by ansyral** [comment]
:shipit:

**Pull request merged**